### PR TITLE
External Link: Use CSS-in-JS

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
This PR updates the `ExternalLink` component to use CSS-in-JS similar to how newer components do like the `AnglePickerControl` and `Text`. This also maintains the existing classname on the `Icon` even though it isn't being used by the component itself, it allows the `Icon` to still be targeted by consumers, making this a non-breaking change.

## How has this been tested?
I've run storybook and observed that there are new visual differences in before and after.

## Screenshots 
Before:
Captura de Tela 2020-09-30 às 10 45 33

After:
Captura de Tela 2020-09-30 às 10 43 53

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/external-link/styles/external-link-styles.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/icons';
+
+export const StyledIcon = styled( Icon )`
+	width: 1.4em;
+	height: 1.4em;
+	margin: -0.2em 0.1em 0;
```
Comment (youknowriad) [inline packages/components/src/external-link/styles/external-link-styles.js:14]: I'm pretty sure these need special care for RTL.

Comment (saramarcondes) [reply packages/components/src/external-link/styles/external-link-styles.js:14]: Does this actually need it? The values as written here will have equal margin for left and right as the CSS values go "top, right, bottom, left". As "left" is omitted, the value for "right" will be used, making it equal on both sides. Is my understanding correct that we don't need to make a change for RTL in this case? If not then I will need some direction from someone more knowledgable than me to know what needs to be changed here.

Comment (youknowriad) [reply packages/components/src/external-link/styles/external-link-styles.js:14]: Seems like you're right here. no particular change is needed here.

**Review by ItsJonQ — APPROVED** [review]
🚀 from me! Thank you @saramarcondes ! Excited to see more components using the CSS-in-JS solutions :)

**Pull request merged**